diff --git a/helper/static/dwm/dwm.c b/helper/static/dwm/dwm.c
index 4717601..5173cc1 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1258,7 +1258,7 @@ movemouse(const Arg *arg)
 			handler[ev.type](&ev);
 			break;
 		case MotionNotify:
-			if ((ev.xmotion.time - lasttime) <= (1000 / 60))
+			if ((ev.xmotion.time - lasttime) <= (1000 / 180))
 				continue;
 			lasttime = ev.xmotion.time;

@@ -1490,7 +1490,7 @@ resizemouse(const Arg *arg)
 			handler[ev.type](&ev);
 			break;
 		case MotionNotify:
-			if ((ev.xmotion.time - lasttime) <= (1000 / 60))
+			if ((ev.xmotion.time - lasttime) <= (1000 / 180))
 				continue;
 			lasttime = ev.xmotion.time;

