diff --git a/blocks.def.h b/blocks.def.h
index dbb3305..b5d7cea 100644
--- a/blocks.def.h
+++ b/blocks.def.h
@@ -1,9 +1,13 @@
+void hello_nix(char* status_buf, int len){
+	strncpy(status_buf, "hello nix ", len);
+}
+
 //Modify this file to change what commands output to your statusbar, and recompile using the make command.
 static const Block blocks[] = {
-	/*Icon*/	/*Command*/		/*Update Interval*/	/*Update Signal*/
-	{"Mem:", "free -h | awk '/^Mem/ { print $3\"/\"$2 }' | sed s/i//g",	30,		0},
-
-	{"", "date '+%b %d (%a) %I:%M%p'",					5,		0},
+	/*Icon*/	/*Command*/	/*C function*/	/*Update Interval*/	/*Update Signal*/
+	{"Mem:", "free -h | awk '/^Mem/ { print $3\"/\"$2 }' | sed s/i//g", NULL,	30,		0},
+	{"", "date '+%b %d (%a) %I:%M%p'",		NULL,			5,		0},
+	{"", NULL,		hello_nix,			5,		0},
 };

 //sets delimiter between status commands. NULL character ('\0') means no delimiter.
diff --git a/dwmblocks.c b/dwmblocks.c
index c8f77b7..0052ccc 100644
--- a/dwmblocks.c
+++ b/dwmblocks.c
@@ -21,6 +21,7 @@
 typedef struct {
 	char* icon;
 	char* command;
+	void (*func)(char *, int);
 	unsigned int interval;
 	unsigned int signal;
 } Block;
@@ -61,11 +62,19 @@ void getcmd(const Block *block, char *output)
 	//make sure status is same until output is ready
 	char tempstatus[CMDLENGTH] = {0};
 	strcpy(tempstatus, block->icon);
-	FILE *cmdf = popen(block->command, "r");
-	if (!cmdf)
+	if (!block->command && !block->func){
 		return;
+	}
 	int i = strlen(block->icon);
-	fgets(tempstatus+i, CMDLENGTH-i-delimLen, cmdf);
+	if (block->command) {
+		FILE *cmdf = popen(block->command, "r");
+		if (!cmdf)
+			return;
+		fgets(tempstatus+i, CMDLENGTH-i-delimLen, cmdf);
+		pclose(cmdf);
+	} else {
+		block->func(tempstatus+i, CMDLENGTH-i-delimLen);
+	}
 	i = strlen(tempstatus);
 	//if block and command output are both not empty
 	if (i != 0) {
@@ -78,7 +87,6 @@ void getcmd(const Block *block, char *output)
 			tempstatus[i++] = '\0';
 	}
 	strcpy(output, tempstatus);
-	pclose(cmdf);
 }

 void getcmds(int time)
diff --git a/helper/static/dwmblocks/Makefile b/helper/static/dwmblocks/Makefile
index 75109ac..d13daa8 100644
--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,7 @@ options:
        @echo "LDFLAGS = ${LDFLAGS}"
        @echo "CC      = ${CC}"

-dwmblocks: dwmblocks.c blocks.def.h blocks.h
+dwmblocks: dwmblocks.c blocks.def.h blocks.h toolkits.h
        ${CC} -o dwmblocks dwmblocks.c ${CFLAGS} ${LDFLAGS}

 blocks.h:
