diff --git a/dwm.c b/dwm.c
index f1d86b2..7f730e5 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1172,7 +1172,7 @@ movemouse(const Arg *arg)
 			handler[ev.type](&ev);
 			break;
 		case MotionNotify:
-			if ((ev.xmotion.time - lasttime) <= (1000 / 60))
+			if ((ev.xmotion.time - lasttime) <= (1000 / 180))
 				continue;
 			lasttime = ev.xmotion.time;
 
@@ -1326,7 +1326,7 @@ resizemouse(const Arg *arg)
 			handler[ev.type](&ev);
 			break;
 		case MotionNotify:
-			if ((ev.xmotion.time - lasttime) <= (1000 / 60))
+			if ((ev.xmotion.time - lasttime) <= (1000 / 180))
 				continue;
 			lasttime = ev.xmotion.time;
 
