diff --git a/plugins/preview-tabbed b/plugins/preview-tabbed
index 8e7751b7..040600b8 100755
--- a/plugins/preview-tabbed
+++ b/plugins/preview-tabbed
@@ -57,10 +57,10 @@ if [ -n "$WAYLAND_DISPLAY" ] ; then
     exit 1
 fi
 
-if type alacritty >/dev/null 2>&1 ; then
-    TERMINAL="alacritty --embed"
-elif type st >/dev/null 2>&1 ; then
+if type st >/dev/null 2>&1 ; then
     TERMINAL="st -w"
+elif type alacritty >/dev/null 2>&1 ; then
+    TERMINAL="alacritty --embed"
 elif type urxvt >/dev/null 2>&1 ; then
     TERMINAL="urxvt -embed"
 elif type xterm >/dev/null 2>&1 ; then
